coalib/core/CircularDependencyError.py

```diff
@@ -1,19 +1,11 @@
-class CircularDependencyError(Exception):
+class CircularDependencyError(RuntimeError):
     """
     An error identifying a circular dependency.
     """
 
-    def __init__(self, node=None):
-        """
-        Creates the CircularDependencyError with a helpful message about the
-        dependency.
-
-        :param node:
-            The node that was encountered twice and closes the dependency
-            circle.
-        """
-        message = 'Circular dependency detected.'
-        if node is not None:
-            message += ' {0!r} -> ... -> {0!r}'.format(node)
-
-        Exception.__init__(self, message)
+    def __init__(self, names=None):
```

Comment:
Please provide a docstring ;) You can use as much as possible from the old one :+1: